pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -171,29 +179,72 @@ func newReconciler(ofClient openflow.Client, ifaceStore interfacestore.Interface
 // invoke the add or update method accordingly.
 func (r *reconciler) Reconcile(rule *CompletedRule) error {
 	klog.Infof("Reconciling rule %s of NetworkPolicy %s/%s", rule.ID, rule.PolicyNamespace, rule.PolicyName)
+	var err error
+	var ofPriority *uint16
 
 	value, exists := r.lastRealizeds.Load(rule.ID)
+	r.priorityMutex.Lock()
```

Comment:
Do we need to lock the whole block? Not just to protect priority assigner?

Reply:
Yes we need to lock the whole block. See https://github.com/vmware-tanzu/antrea/pull/803#discussion_r442631932. I've tested that only locking the priorityAssigner will cause agent-restart case to fail assigning correct priorities.